### PR TITLE
added update/import for device resource

### PR DESCRIPTION
**Pull request by shenda1**
Added update/import for device resource

image

### CODE REVIEW

File: powerflex/device_resource.go [hunk 1]
```diff
@@ -401,6 +349,122 @@ func (r *deviceResource) Read(ctx context.Context, req resource.ReadRequest, res
...
+	}
+
+	sp, err := getStoragePoolType(r.client, plan.StoragePoolID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error getting storage pool instance with ID: "+plan.StoragePoolID.ValueString(),
+			"unexpected error: "+err.Error(),
+		)
+		return
+	}
```
Comment (rounak-adhikary) [inline powerflex/device_resource.go:383]: Can we return the storagepool from the `r.getStoragePoolID(&plan)` function?
We do not really need to make the api calls on line 376.

File: powerflex/device_resource.go [hunk 2]
```diff
@@ -401,6 +349,122 @@ func (r *deviceResource) Read(ctx context.Context, req resource.ReadRequest, res
...
+		}
+	}
+
+	if plan.DevicePath.ValueString() != state.DevicePath.ValueString() {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("device_path"),
+			"Please update the device path in sync with current path of device.",
+			"Please update the device path in sync with current path of device.",
+		)
+	}
```
Comment (rounak-adhikary) [inline powerflex/device_resource.go:439]: Sooo... `device_path` cannot be updated?

File: powerflex/device_resource.go [hunk 3]
```diff
@@ -401,6 +349,122 @@ func (r *deviceResource) Read(ctx context.Context, req resource.ReadRequest, res
...
+	// Update original path if there is change in the current path
+	if state.DevicePath.ValueString() != state.DeviceOriginalPath.ValueString() {
+		err = sp.UpdateDeviceOriginalPathways(state.ID.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error updating device pathway with ID: "+plan.ID.ValueString(),
+				err.Error(),
+			)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (rounak-adhikary) [reply powerflex/device_resource.go:450]: if deviceOriginalPathway is not an independent attribute of a device, do we need to support that field?

File: powerflex/device_resource.go (hunk 2 above)
Comment (shenda1) [reply powerflex/device_resource.go:439]: As discussed, the current path of the device can't be changed.  If there is any change in the device path, the current path of the device gets changed automatically.

File: powerflex/device_resource.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (shenda1) [reply powerflex/device_resource.go:450]: There are 2 path-related attributes for the device i.e. currentPath and originalPath. If there is a change in the path due to SDS restart, the current path gets changed automatically but not originalPath. OriginalPath will be changed only after sp.UpdateDeviceOriginalPathways() is executed successfully.

File: powerflex/device_resource.go (hunk 1 above)
Comment (shenda1) [reply powerflex/device_resource.go]: done

File: powerflex/device_resource.go [hunk 4]
```diff
@@ -68,9 +69,9 @@ func (r *deviceResource) Schema(_ context.Context, _ resource.SchemaRequest, res
 				},
 			},
 			"device_path": schema.StringAttribute{
```
Comment (rounak-adhikary) [inline powerflex/device_resource.go:71]: Please add that it cannot be updated.

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Pull request merged**